chrome/content/zotero/xpcom/quickCopy.js

```diff
@@ -308,7 +308,44 @@ Zotero.QuickCopy = new function() {
 		
 		throw ("Invalid mode '" + format.mode + "' in Zotero.QuickCopy.getContentFromItems()");
 	};
-	
+
+	/**
+	 * Generate a note item from an array of annotations.
+	 * This is a workaround to be able to translate annotations in getContentFromItems
+	 * even though annotations do not have a cslType.
+	 * @param {Zotero.Item|Object[]} annotations - Array of JSON annotations or Zotero.Item annotations.
+	 * If an array of Zotero.Items is passed, they will be converted to JSON.
+	 * @returns {Zotero.Item} - A note item with the annotations serialized as HTML.
+	 */
+	this.wrapAnnotationsAsNote = function (annotations) {
+		let jsonAnnotations = [];
+		for (let annotation of annotations) {
+			if (annotation instanceof Zotero.Item) {
```

Comment:
When would these not be `Zotero.Item` objects? Can we not just require that?

Reply:
In reader `onSetDataTransferAnnotations`, the passed annotations are already converted to JSON, so we just use those objects right away. But in case of drag-drop of annotation rows or "Copy Annotations" menu option, I thought it's better to be able to pass the actual `Zotero.Item`, so we don't  have to also call `Zotero.Annotations.toJSONSync` and set `attachmentItemID` before each `wrapAnnotationsAsNote`.